src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1694,15 +1701,67 @@ public void onRequestCancelled(@NotNull final IToken token)
     @NotNull
     @Override
     public ITextComponent getDisplayName(@NotNull final IToken token)
+    {
+        return new TextComponentString(getCitizenForRequest(token).map(CitizenData::getName).orElse("<UNKNOWN>"));
+    }
+
+    public Optional<CitizenData> getCitizenForRequest(@NotNull final IToken token)
     {
         if (!requestsByCitizen.containsKey(token))
         {
-            return new TextComponentString(this.getSchematicName() + " <UNKNOWN>");
+            return Optional.empty();
         }
 
         Integer citizenData = requestsByCitizen.get(token);
-        return new TextComponentString(this.getSchematicName() + " " + getColony().getCitizen(citizenData).getName());
+        return Optional.of(getColony().getCitizen(citizenData));
     }
 
+
     //------------------------- !END! RequestSystem handling for minecolonies buildings -------------------------//
+
+    //------------------------- !Start! Capabilities handling for minecolonies buildings -------------------------//
+
+    @Override
+    public boolean hasCapability(
+      @Nonnull final Capability<?> capability, @Nullable final EnumFacing facing)
+    {
+        if (capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY && facing == null)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1126)